deploy/eks/kiam-agent-deamonset.yaml

```diff
@@ -0,0 +1,82 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  namespace: kube-system
+  name: kiam-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: kiam
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9620"
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      labels:
+        k8s-app: kiam
+        role: agent
+    spec:
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      nodeSelector:
+        kubernetes.io/role: node
+      volumes:
+        - name: ssl-certs
+          hostPath:
+            # for AWS linux or RHEL distros
+            path: /etc/pki/ca-trust/extracted/pem/
+            #path: /usr/share/ca-certificates
+        - name: tls
+          secret:
+            secretName: kiam-agent-tls
+        - name: xtables
+          hostPath:
+            path: /run/xtables.lock
+      containers:
+        - name: kiam
+          securityContext:
+            privileged: true
+            capabilities:
+              add: ["NET_ADMIN"]
+          image: quay.io/uswitch/kiam:v2.8 # USE A TAGGED RELEASE IN PRODUCTION
+          imagePullPolicy: Always
+          resources:
+            limits:
+              memory: 200Mi
+              cpu: 200m
+            requests:
+              cpu: 10m
+              memory: 24Mi
+          command:
+            - /agent
+          args:
+            - "--iptables"
+            - "--host-interface=cali+"
```

Comment:
not eni+ ?

Reply:
https://github.com/uswitch/kiam#typical-cni-interface-names also suggested using `!eth0`.

If someone running in EKS could verify that'd be great- we're a way off testing it out.

Reply:
@pingles 
Agent is not up with either one. I used eni+ in kube2iam.
Container is backoff.
  Warning  BackOff                1m (x6 over 1m)  kubelet, ip-10-55-5-39.us-west-2.compute.internal  Back-off restarting failed container

Reply:
`!eth0` fails for me on a `t3.large` instance.

The [AWS CNI plugin](https://github.com/aws/amazon-vpc-cni-k8s) will attach [up to 15 network adapters](https://github.com/aws/amazon-vpc-cni-k8s/blob/b1bfcb44fb7dc1aba9bfd2a273ec1954c391ee9d/pkg/awsutils/vpc_ip_resource_limit.go#L96) to a worker node depending on the worker node instance size.  The instance size also determines how many IP addresses each network adapter gets, which could be [as many as 30 IPs](https://github.com/aws/amazon-vpc-cni-k8s/blob/b1bfcb44fb7dc1aba9bfd2a273ec1954c391ee9d/pkg/awsutils/vpc_ip_resource_limit.go#L204) per adapter.

The AWS CNI plugin will allocate an IP address for each running pod, so if you listen on `!eth0` then any pod with an IP assigned to the `eth0` interface cannot be detected by kiam-agent.  The assumption https://github.com/uswitch/kiam#typical-cni-interface-names makes is that the `eth0` IPs are limited to kube-system pods when they're not.  Naturally, this is a problem.  

One solution would be to set the host interface to `!eth15` since there will never be 16 network adapters, but this is just kicking the can down the road since AWS could create a new image which can have 16 network adapters on it.

The better solution is to stop using AWS CNI plugin.  I opted for weave over calico.

Reply:
`!eth0` fails for me on a `t3.large` instance.

The [AWS CNI plugin](https://github.com/aws/amazon-vpc-cni-k8s) will attach [up to 15 network adapters](https://github.com/aws/amazon-vpc-cni-k8s/blob/b1bfcb44fb7dc1aba9bfd2a273ec1954c391ee9d/pkg/awsutils/vpc_ip_resource_limit.go#L96) to a worker node depending on the worker node instance size.  The instance size also determines how many IP addresses each network adapter gets, which could be [as many as 30 IPs](https://github.com/aws/amazon-vpc-cni-k8s/blob/b1bfcb44fb7dc1aba9bfd2a273ec1954c391ee9d/pkg/awsutils/vpc_ip_resource_limit.go#L204) per adapter.

The AWS CNI plugin will allocate an IP address for each running pod, so if you listen on `!eth0` then any pod with an IP assigned to the `eth0` interface cannot be detected by kiam-agent.  The assumption https://github.com/uswitch/kiam#typical-cni-interface-names makes is that the `eth0` IPs are limited to kube-system pods when they're not.  Naturally, this is a problem.  

One solution would be to set the host interface to `!eth15` since there will never be 16 network adapters, but this is just kicking the can down the road since AWS could create a new image which can have 16 network adapters on it.

The better solution is to stop using AWS CNI plugin.  I opted for weave over calico.

Reply:
can it not be like .*?

Reply:
I thought for EKS the interface was `eni+`?

Reply:
eni+ worked for me on EKS